src/moscot/problems/base/_mixins.py

```diff
@@ -476,3 +478,100 @@ def _cell_aggregation_transition(
             tm = pd.concat([tm, to_app], verify_integrity=True, axis=0)
             df_1 = df_1.drop(current_cells, axis=1)
         return tm
+
+    # adapted from CellRank (github.com/theislab/cellrank)
+    def compute_feature_correlation(
+        self: AnalysisMixinProtocol[K, B],
+        obs_key: str,
+        method: Literal["fischer", "perm_test"] = CorrTestMethod.FISCHER,
+        annotation: Optional[Dict[str, List[str]]] = None,
+        layer: Optional[str] = None,
+        features: Optional[Union[List[str], Literal["human", "mouse", "drosophila"]]] = None,
+        confidence_level: float = 0.95,
+        n_perms: int = 1000,
+        seed: Optional[int] = None,
+        **kwargs: Any,
+    ) -> pd.DataFrame:
+        """
+        Compute correlation of push or pull distribution with features.
+
+        Correlates (processed) count data with probabilities of cells mapped to a subset of cells, e.g. for a
+        certain celltype.
+
+        Parameters
+        ----------
+        obs_key
+            Column of :attr:`adata.obs` containing push-forward or pull-back distributions.
+        method
+            Mode to use when calculating p-values and confidence intervals. Valid options are:
+                - `{CorrTestMethod.FISCHER!r}` - use Fischer transformation :cite:`fischer:21`.
+                - `{CorrTestMethod.PERM_TEST!r}` - use permutation test.
+        annotation
+            If not `None`, this defines the subset of data to be considered when computing the correlation.
+            Its key should correspond to a key in
+            :attr:`anndata.AnnData.obs` and its value to a list containing a subset of categories present in
+            :attr:`anndata.AnnData.obs` ``['{annotation.keys()[0]}']``.
+        layer
+            Key from :attr:`anndata.AnnData.layers` from which to get the expression.
+            If `None`, use :attr:`anndata.AnnData.X`.
+        features
+            Features of :class:`anndata.AnnData` which the correlation of ``anndata.AnnData.obs['{obs_key}']`` is
+            computed with:
+            - If `None`, all features will be taken into account.
+            - If of type :obj:`list`, features from :attr:`anndata.AnnData.var_names` will be taken.
+            - If `human`, `mouse`, or `drosophila`, the features are subsetted to transcription factors.
+        confidence_level
+            Confidence level for the confidence interval calculation. Must be in interval `[0, 1]`.
+        n_perms
+            Number of permutations to use when ``method = {CorrTestMethod.PERM_TEST!r}``.
```

Comment:
Change the ``method = {CorrTestMethod.PERM_TEST!r}`` please, it's no longer extrapolated (it was in my code since I was using the decorator for this).